packages/fxa-payments-server/src/components/CouponForm/index.tsx

```diff
@@ -106,6 +113,9 @@ export const CouponForm = ({
     coupon ? coupon.promotionCode : ''
   );
   const [error, setError] = useState<CouponErrorMessageType | null>(null);
+  const [errorMessage, setErrorMessage] = useState<CouponErrorMessage | string>(
```

Comment:
Why do we need a `setError` and a `setErrorMessage`? IIUC `error` is the l10n id and `errorMessage` is the fallback message. Can we use `error` to key into a default error message in `getDefaultErrorMessage`? I see you are doing something more similar to that for `PaymentErrorView` below.

Reply:
Thank you! Agreed. I will make this change.